File: pact-jvm-model/src/main/java/au/com/dius/pact/model/PactSpecVersion.java
package au.com.dius.pact.model;

/**
 * Pact Specification Version
 */
public enum PactSpecVersion {
    V1(2), V1_1(2), V2(2), V3(3);

    private final int supportedVersion;

    PactSpecVersion(int supportedVersion) {
        this.supportedVersion = supportedVersion;
    }

    public int getSupportedVersion() {
        return supportedVersion;
    }

    public static PactSpecVersion fromInt(int version) {
        switch (version) {
            case 1:
                return V1;
            case 3:
                return V3;
        }

        return V2;
    }
}

File: pact-jvm-model/src/main/kotlin/au/com/dius/pact/model/PactSpecVersion.kt
package au.com.dius.pact.model

/**
 * Pact Specification Version
 */
@Suppress("NamingConventionViolation")
enum class PactSpecVersion {
  V1, V1_1, V2, V3;

  companion object {
    @JvmStatic
    fun fromInt(version: Int): PactSpecVersion {
      return when (version) {
        1 -> V1
        2 -> V2
        else -> V3
      }
    }
  }
}
